### DISCUSSION TITLE
ServerMiddleware how to abort request?

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#17804] [Questions]
How to abort a request while the middleware is running?
What i have now:
`@/server/middleware/auth-middleware.ts`
```ts
import { TokenService } from '../services/token-service';
import type { IncomingMessage, ServerResponse } from 'http';
import { useCookie } from 'h3';

const INCLUDE_URL: string[] = ['/api/get-users'];

export default async (req: IncomingMessage, res: ServerResponse) => {
	console.log('Auth middleware URL: ', req.url);
	if (!INCLUDE_URL.includes(req.url)) {
		return;
	}

	try {
		const accessToken = useCookie(req, 'accessToken');
		console.log(`Auth MiddleWare accessToken: ${accessToken}`);

		if (!accessToken) {
			throw new Error('Отсутствует accessToken');
		}

		const userData = await TokenService.verifyAccessToken(accessToken);
		if (!userData) {
			throw new Error('Невалидный accessToken');
		}
	} catch (e) {
		console.log(`[Auth Middleware] ${e.message}`);
		res.statusCode = 401;
		res.end(JSON.stringify({ message: e.message }));
		return;
	}
	return;
};
```
As soon as the exception is triggered - my site dies

**Comment by lukas-pierce** [comment]
I think you forgot call next() callback which comes as third argument in your server middleware function, read the [nuxt documentation](https://nuxtjs.org/docs/configuration-glossary/configuration-servermiddleware#custom-server-middleware):

<img width="737" alt="image" src="https://user-images.githubusercontent.com/4938316/159171445-c7dfd556-dc9e-46f7-a4b9-ed5ad0faac5c.png">

**Reply by mihail727** [reply]
But the documentation for nuxt 3 says nothing about this.
https://v3.nuxtjs.org/docs/directory-structure/server#server-middleware

**Reply by lukas-pierce** [reply]
I think you should to check arguments count, for this purposes use function with context instead lambda to access the function's `arguments`:

```js
export default async function (req: IncomingMessage, res: ServerResponse) {
   console.log('arguments count:', arguments.length); // If there is 3 then it means third is next callback
}
```

And also deeper discover v3 docs: 
https://v3.nuxtjs.org/docs/directory-structure/nuxt.config#servermiddleware

**Reply by lukas-pierce** [reply]
Just try to add third arg `next` and replace your early returns with `return next()`
